xbmc/video/dialogs/GUIDialogVideoBookmarks.cpp

```diff
@@ -235,6 +239,30 @@ void CGUIDialogVideoBookmarks::OnRefreshList()
     item->SetArt("thumb", m_bookmarks[i].thumbNailImage);
     m_vecItems->Add(item);
   }
+  // add chapters if around
+  for (int i=1;i<=g_application.m_pPlayer->GetChapterCount();++i)
+  {
+    std::string chapterName;
+    g_application.m_pPlayer->GetChapterName(chapterName, i);
+    if (chapterName.empty())
+      chapterName = StringUtils::Format(g_localizeStrings.Get(25010).c_str(), i);
+    int64_t pos = g_application.m_pPlayer->GetChapterPos(i);
+    std::string time = 
+      StringUtils::SecondsToTimeString((long) pos, TIME_FORMAT_HH_MM_SS);
+    std::string name = StringUtils::Format("%s (%s)",
+                                           chapterName.c_str(), time.c_str());
+    CFileItemPtr item(new CFileItem(name));
+    time = StringUtils::Format("chapter://%s/%i", path.c_str(), i);
+    std::string cachefile = CTextureCache::Get().GetCachedPath(CTextureCache::Get().GetCacheFile(time)+".jpg");
+    if (XFILE::CFile::Exists(cachefile))
+      item->SetArt("thumb", cachefile);
+    else if (createThumbJob)
```

Comment:
assume yo have 10 chapters. at the first call to RefreshList is schedules 10 jobs. When the fist job finishes, it calls RefereshList again. At this time it will schedule another 9 jobs.